File: fjt.api/models/RFQAssyQuoteSubmittedAssyDetail.js
const Sequelize = require('sequelize');

module.exports = (sequelize) => {
    const RFQAssyQuoteSubmittedAssyDetail = sequelize.define('RFQAssyQuoteSubmittedAssyDetail', {
        id: {
            allowNull: false,
            autoIncrement: true,
            primaryKey: true,
            type: Sequelize.INTEGER
        },
        refSubmittedQuoteID: {
            allowNull: false,
            type: Sequelize.INTEGER
        },
        rfqAssyID: {
            type: Sequelize.INTEGER,
            allowNull: false
        },
        qty: {
            type: Sequelize.INTEGER,
            allowNull: false
        },
        turnTime: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        turnType: {
            type: Sequelize.STRING,
            allowNull: false,
            validate: {
                notEmpty: true,
                len: [1, 1]
            }
        },
        materialCost: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        materialLeadTime: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        laborCost: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        laborLeadTime: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        additionalCost: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        customItemLeadTime: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        unitPrice: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        totalLeadTime: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        extendedCost: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        excessMaterialCost: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        additionalCostDetail: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        isDeleted: {
            type: Sequelize.BOOLEAN,
            allowNull: true
        },
        createdBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        updatedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        deletedBy: {
            type: Sequelize.STRING,
            allowNull: true
        },
        createdAt: {
            type: Sequelize.DATE,
            allowNull: true
        },
        updatedAt: {
            type: Sequelize.DATE,
            allowNull: true
        },
        deletedAt: {
            type: Sequelize.DATE,
            allowNull: true
        },
        createByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        updateByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        deleteByRoleId: {
            type: Sequelize.INTEGER,
            allowNull: true
        },
        allCost: {
            type: Sequelize.DECIMAL(16, 6),
            allowNull: false
        },
        allDays: {
            type: Sequelize.INTEGER,
            allowNull: false
        }
    }, {
        paranoid: true,
        tableName: 'rfq_assy_quote_submitted_assyDetail'
    });

    RFQAssyQuoteSubmittedAssyDetail.associate = (models) => {
        RFQAssyQuoteSubmittedAssyDetail.belongsTo(models.ECOTypeCategory, {
            as: 'termsAndConditionCategory',
            foreignKey: 'termsconditionCatID'
        });
        RFQAssyQuoteSubmittedAssyDetail.belongsTo(models.ECOTypeValues, {
            as: 'termsAndConditionTypeValues',
            foreignKey: 'termsconditionTypeValueID'
        });
        RFQAssyQuoteSubmittedAssyDetail.belongsTo(models.RFQAssembliesQuotationSubmitted, {
            as: 'rfqAssyQuoteSubmitted',
            foreignKey: 'RefSubmittedQuoteID'
        });
    };

    return RFQAssyQuoteSubmittedAssyDetail;
};

File: FJT.Reporting/Models/RFQ_Assy_Quote_Submitted_Assydetail.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FJT.Reporting.Models
{
    [Table("rfq_assy_quote_submitted_assydetail")]
    public class RFQ_Assy_Quote_Submitted_Assydetail
    {
        [Key]
        public int id { get; set; }
        public int? refSubmittedQuoteID { get; set; }
        public int? rfqAssyID { get; set; }
        public int? qty { get; set; }
        public int? turnTime { get; set; }
        public string turnType { get; set; }
        public decimal? materialCost { get; set; }
        public decimal? materialLeadTime { get; set; }
        public decimal? laborCost { get; set; }
        public decimal? laborLeadTime { get; set; }
        public decimal? additionalCost { get; set; }
        public decimal? customItemLeadTime { get; set; }
        public decimal? unitPrice { get; set; }
        public decimal? totalLeadTime { get; set; }
        public decimal? extendedCost { get; set; }
        public decimal? excessMaterialCost { get; set; }
        public decimal? additionalCostDetail { get; set; }
        public bool isDeleted { get; set; }
        public string createdBy { get; set; }
        public DateTime createdAt { get; set; }
        public DateTime updatedBy { get; set; }
        public string updatedAt { get; set; }
        public string deletedBy { get; set; }
        public DateTime deletedAt { get; set; }
        public decimal? nreCost { get; set; }
        public decimal? nreDays { get; set; }
        [NotMapped]
        public int? turntimedays { get; set; }
        public string nretoolingdescription { get; set; }
        public string priceGroup { get; set; }
    }
}